### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by mrocklin**

### CODE REVIEW

File: castra/core.py [hunk 1]
```diff
@@ -152,7 +156,7 @@ def extend(self, df):
         # TODO: Ensure that df is consistent with existing data
         if not df.index.is_monotonic_increasing:
             df = df.sort_index(inplace=False)
-        if len(self.partitions) and df.index[0] < self.partitions.index[0]:
+        if len(self.partitions) and df.index[0] <= self.partitions.index[0]:
```
Comment (jcrist) [inline castra/core.py]: Shouldn't those `0`s should be `-1`s - we want to make sure this is after the most recent partition, not after the first partition.

Comment (mrocklin) [inline castra/core.py]: Indeed.  Fixing.

Comment (mrocklin) [inline castra/core.py]: Fixed.

**Comment by jcrist** [comment]
lgtm.

**Pull request merged**